libsolidity/ast/Types.h

```diff
@@ -773,6 +773,36 @@ class ArrayType: public ReferenceType
 	mutable boost::optional<TypeResult> m_interfaceType_library;
 };
 
+class ArraySliceType: public ReferenceType
+{
+public:
+	explicit ArraySliceType(ArrayType const& _arrayType): ReferenceType(_arrayType.location()), m_arrayType(_arrayType) {}
+	Category category() const override { return Category::ArraySlice; }
+
+	BoolResult isImplicitlyConvertibleTo(Type const& _other) const override;
+	std::string richIdentifier() const override;
+	bool operator==(Type const& _other) const override;
+	unsigned calldataEncodedSize(bool) const override { solAssert(false, ""); }
+	unsigned calldataEncodedTailSize() const override { return 32; }
+	bool isDynamicallySized() const override { return true; }
+	bool isDynamicallyEncoded() const override { return true; }
+	u256 storageSize() const override { return 1; }
```

Comment:
Is this meaningful?

Reply:
It's what dynamically sized arrays (of any kind) do - that's why it's here, since array slices are basically dynamically sized arrays with a special stack representation...

Reply:
I removed it in any case - returning 1 is the default implementation anyways...